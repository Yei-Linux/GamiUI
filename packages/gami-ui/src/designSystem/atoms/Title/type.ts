import { IComponentsVariant } from 'core/domain/interfaces/IComponentsVariant'
import { LevelType } from 'core/domain/types'
import {
  TComponentEvents,
  TDesignTokens,
  TDesignUI,
} from 'core/domain/types/design-system'

export type TTitle = {
  /**
   * Level Title
   */
  level?: LevelType
  /**
   * Title Text Content
   */
  children: React.ReactNode
}

export type TTitleComponent = TTitle &
  TDesignTokens &
  TDesignUI &
  TComponentEvents<any, any> &
  IComponentsVariant
