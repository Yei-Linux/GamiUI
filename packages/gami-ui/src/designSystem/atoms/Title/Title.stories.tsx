import React from 'react'
import { Meta } from '@storybook/react'

import Title from '.'

export default {
  title: 'Atoms/Title',
  component: Title,
  args: {},
  argTypes: {},
} as Meta

export const Basic = (args: any) => <Title {...args}>Titulo h1</Title>
Basic.args = {
  margin: 'xs:0 0 1rem 0',
  padding: 'xs:0 0 0 0',
}
