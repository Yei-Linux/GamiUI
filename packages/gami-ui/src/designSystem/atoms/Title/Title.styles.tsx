import styled from '@emotion/styled'
import { GlobalStylesComponent } from 'styles/utilities/commonComponent'

export const Title = styled(GlobalStylesComponent('div'))``
