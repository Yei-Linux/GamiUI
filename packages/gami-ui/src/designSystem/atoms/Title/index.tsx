export { default } from './Title'
export { options } from './constants'
export { argTypes } from './storyConfig'
