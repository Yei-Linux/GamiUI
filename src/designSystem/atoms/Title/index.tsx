export { default } from './Title'
export { options } from './constants'
