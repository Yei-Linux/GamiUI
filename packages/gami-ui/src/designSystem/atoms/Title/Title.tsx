import React from 'react'
import { IGeneralProps } from 'core/domain/interfaces/IGeneralProps'
import * as S from './Title.styles'
import { getGenericPropStyles } from 'styles/utilities/genericPropStyles'
import withDefaults from 'hocs/WithDefault'
import useCssHandle from 'hooks/useCssHandle'
import { cls } from 'core/utils/cls'
import { LevelType } from 'core/domain/types'

export interface ITitle extends IGeneralProps {
  /**
   * Level Title
   */
  level?: LevelType
  /**
   * Title Text Content
   */
  children: React.ReactNode
}

const Title = React.forwardRef(
  ({ level = 'h1', children, ...genericsProps }: ITitle, ref) => {
    const { handles } = useCssHandle({
      classes: {
        wrapper: ['wrapper'],
      },
      componentPrefixCls: 'title',
      customPrexiCls: genericsProps.className,
    })

    return (
      <S.Title
        ref={ref}
        className={cls(handles.wrapper, genericsProps?.className ?? '')}
        as={level}
        {...getGenericPropStyles(genericsProps)}
      >
        {children}
      </S.Title>
    )
  }
)

const defaultProps = {
  shadow: 'none',
  width: 'auto',
  height: 'auto',
}

Title.displayName = 'Title'

type TitleComponent<P> = React.NamedExoticComponent<P> & {
  defaultProps: P
}

export default withDefaults(Title, defaultProps) as TitleComponent<ITitle>
