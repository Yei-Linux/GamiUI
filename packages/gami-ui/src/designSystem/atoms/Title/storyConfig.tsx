import { IStoryMainConfig } from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
  madegenericPropsControl,
} from 'core/helpers/storybook.helper'
import Title, { options } from '.'

const genericArgTypes = madegenericPropsControl([])

const storyArgTypes = {
  level: {
    control: { type: 'select', options: options.typeLevel },
    description: 'Level Title',
    table: {
      type: { summary: 'string' },
      defaultValue: Title.defaultProps?.level,
    },
  },
  ...genericArgTypes,
}

const docArgTypes = {
  children: {
    control: 'object',
    description: 'Title Text Content',
    table: {
      type: { summary: 'React.ReactNode' },
    },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/Title ðŸŸ¢',
  component: Title,
  args: {},
  argTypes: storyArgTypes,
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: {
    args: {
      children: 'Gami Title',
      fontWeight: 'semibold',
      level: 'h4',
      textAlign: 'center',
      shadow: 'sm',
      padding: '10px',
    },
  },
  shadow: {
    args: {
      children: 'Gami Title',
      fontWeight: 'semibold',
      level: 'h4',
      textAlign: 'center',
      shadow: 'sm',
      padding: '10px',
    },
  },
  width: {
    args: {
      children: 'GamiUI',
      fontWeight: 'semibold',
      level: 'h4',
      textAlign: 'center',
      shadow: 'sm',
      padding: '10px',
    },
  },
  height: {
    args: {
      children: 'GamiUI',
      fontWeight: 'semibold',
      level: 'h4',
      textAlign: 'center',
      shadow: 'sm',
      padding: '10px',
    },
  },
  size: {
    args: {
      children: 'GamiUI',
      fontWeight: 'semibold',
      level: 'h4',
      textAlign: 'center',
      shadow: 'sm',
      padding: '10px',
    },
  },
  fontWeight: {
    args: {
      children: 'GamiUI',
      textAlign: 'center',
      shadow: 'sm',
      padding: '10px',
    },
  },
  textAlign: {
    args: {
      width: 'lg',
      children: 'GamiUI',
      level: 'h4',
      shadow: 'sm',
      padding: '10px',
    },
  },
  margin: {
    args: {
      children: 'Gami Title',
      level: 'h4',
      textAlign: 'center',
      shadow: 'sm',
      padding: '10px',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
  padding: {
    args: {
      children: 'Gami Title',
      level: 'h4',
      textAlign: 'center',
      shadow: 'sm',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px'],
  },
})

const storiesComponent = [
  {
    storyName: 'WithChildren ðŸ™‚',
    self: {
      args: {},
      variants: {
        examples: [
          {
            label: 'Title A',
            value: 'Text A',
          },
          {
            label: 'Title B',
            value: 'Text B',
          },
        ],
        field: 'children',
      },
    },
  },
  {
    storyName: 'WithLevels ðŸ™‚',
    self: {
      args: { children: 'GamiUI Text' },
      variants: {
        examples: options.typeLevel.map((level) => ({
          label: level,
          value: level,
        })),
        field: 'level',
      },
    },
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: Title,
})

export { storyConfig, argTypes }
