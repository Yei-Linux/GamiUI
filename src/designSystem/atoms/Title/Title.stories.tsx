import React from 'react'
import { Meta } from '@storybook/react'

import Title from '.'

export default {
  title: 'Atoms/Title',
  component: Title,
  args: { ...Title.defaultProps },
  argTypes: {},
} as Meta

export const Basic = (args: any) => <Title {...args}>Titulo h1</Title>
Basic.args = {}
