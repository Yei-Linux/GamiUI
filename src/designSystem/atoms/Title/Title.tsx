import React from 'react'
import { IGeneralProps } from 'core/domain/interfaces/IGeneralProps'
import * as S from './Title.styles'
import { getGenericPropStyles } from 'styles/utilities/genericPropStyles'

export interface ITitle extends IGeneralProps {
  /**
   * Tag Title
   */
  level?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

  /**
   * Title Text Content
   */
  children: React.ReactNode
}

const Title = ({ level = 'h1', children, ...genericsProps }: ITitle) => {
  return (
    <S.Title as={level} {...getGenericPropStyles(genericsProps)}>
      {children}
    </S.Title>
  )
}

export default Title
