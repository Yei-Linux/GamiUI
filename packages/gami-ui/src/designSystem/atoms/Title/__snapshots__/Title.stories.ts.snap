// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Title Basic 1`] = `
<div
  className="css-1ejwpna-TitleWrapper e5aka180"
>
  <h1>
    Titulo h1
  </h1>
</div>
`;
